pennylane/math/utils.py

```diff
@@ -241,13 +244,23 @@ def requires_grad(tensor):
 
         return should_record_backprop([tf.convert_to_tensor(tensor)])
 
-    if interface in ("torch", "autograd"):
-        return tensor.requires_grad
+    if interface == "autograd":
+        if isinstance(tensor, ArrayBox):
+            return True
+
+        # Currently, in the Autograd interface, we assume
+        # that all objects are differentiable by default.
+        return getattr(tensor, "requires_grad", True)
+
+    if interface == "torch":
+        return getattr(tensor, "requires_grad", False)
 
     if interface == "numpy":
         return False
 
     if interface == "jax":
-        return True
+        import jax

```

Comment:
There is probably no other way, but won't code reload the library every time we build a tape here?

Reply:
Nope! Python will only import the library the first time the import statement is called, so there is no import overhead (https://stackoverflow.com/questions/32151193/is-there-a-performance-cost-putting-python-imports-inside-functions)